UnityProject/Storybook/Assets/Code/Scripts/Game/Player/PlayerMover.cs

```diff
@@ -0,0 +1,338 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+/// <summary>
+/// Class for handling player movement.
+/// </summary>
+public class PlayerMover : NetworkMover
+{
+
+    private RoomObject m_currentRoom;
+    private RoomData m_currentRoomData;
+    private Location m_currentRoomLoc;
+    private List<NetworkNodeMover> m_worldPlayers = new List<NetworkNodeMover>();
+    private int m_leaderId = 1;
+    private bool m_areButtonsEnabled = false;
+
+    /// <summary>
+    /// Spawns all the players in the given room object
+    /// Called only to place the players in the starting room of the floor
+    /// </summary>
+    /// <param name="room">The room to spawn the players in</param>
+    public void SpawnInRoom(RoomObject room)
+    {
+        m_currentRoom = room;
+        m_currentRoomLoc = room.RoomLocation;
+        MapManager mapManager = FindObjectOfType<MapManager>();
+        m_currentRoomData = mapManager.GetRoomData(room.RoomLocation.X, room.RoomLocation.Y);
+
+        photonView.RPC("SendRoomData", PhotonTargets.Others, m_currentRoomData.IsNorthDoorActive, m_currentRoomData.IsEastDoorActive,
+            m_currentRoomData.IsSouthDoorActive, m_currentRoomData.IsWestDoorActive, m_currentRoomData.RoomType);
+        photonView.RPC("SendRoomLoc", PhotonTargets.Others, m_currentRoomLoc.X, m_currentRoomLoc.Y);
+
+        foreach(NetworkNodeMover player in m_worldPlayers)
+        {
+            player.IsAtTarget = true;
+        }
+
```

Comment:
First, all the NetworkNodeMovers for player world pawns can be handled mostly through the editor, you don't need all this code. Secondly, this class should not be managing the player pawns this much. The idea of the player mover is that it moves around from room to room. The players models are children of it so they move as well. At this point that is really all the logic we are looking for. The only logic that's needed to make that work is spawning the players world pawn which would probably be handled by another class for when the player is spawned.